### ISSUE TITLE
3.17.0 - Edit message stopped working - Internal Server Error

### ISSUE THREAD

**Issue by urosino** [RocketChat/Rocket.Chat#22830]
### Description:
You can edit message, but members won't see changes.

### Steps to reproduce:
1. Write something.
2. Then edit.

### Expected behavior:
1. To be able to edit messages.

### Actual behavior:
The "Internal Server Error" popup appears, and the message is not updated.
![error](https://user-images.githubusercontent.com/25570106/127848998-c257562f-4e39-449f-97d9-7c3d4bf2dc55.png)

### Server Setup Information:
* Version of Rocket.Chat Server: 3.17.0
* Operating System:  Ubuntu 18.04
* Deployment Method: Docker image
* Number of Running Instances: 1
* DB Replicaset Oplog: rs0
* NodeJS Version: 12.18.4
* MongoDB Version: 4.0.12 / mmapv1

### Client Setup Information
- Desktop App or Browser Version: Desktop App, 3.2.3

- Operating System: Windows 10

### Additional context
 - No additional context.

### Relevant logs:
```
Exception while invoking method 'updateMessage' TypeError: Cannot read property 't' of undefined
    at isOmnichannelRoom (definition/IRoom.ts:96:90)
    at ee/app/canned-responses/server/hooks/onMessageSentParsePlaceholder.ts:32:7
    at callbacks.runItem (app/callbacks/lib/callbacks.js:111:70)
    at Object.callbacks.runItem (app/metrics/server/callbacksMetrics.js:24:20)
    at app/callbacks/lib/callbacks.js:39:35
    at app/callbacks/lib/callbacks.js:45:45
    at callbacks.run (app/callbacks/lib/callbacks.js:127:9)
    at Object.callbacks.run (app/metrics/server/callbacksMetrics.js:14:17)
    at updateMessage (app/lib/server/functions/updateMessage.js:48:22)
    at MethodInvocation.updateMessage (app/lib/server/methods/updateMessage.js:71:10)
    at MethodInvocation.methodsMap.<computed> (app/lib/server/lib/debug.js:76:34)
    at maybeAuditArgumentChecks (packages/ddp-server/livedata_server.js:1771:12)
    at packages/ddp-server/livedata_server.js:719:19
    at Meteor.EnvironmentVariable.EVp.withValue (packages/meteor.js:1234:12)
    at packages/ddp-server/livedata_server.js:717:46
    at Meteor.EnvironmentVariable.EVp.withValue (packages/meteor.js:1234:12)
    at packages/ddp-server/livedata_server.js:715:46
    at new Promise (<anonymous>)
    at Session.method (packages/ddp-server/livedata_server.js:689:23)
    at packages/ddp-server/livedata_server.js:559:43
server.js:204 Meteor ➔ method UserPresence:away -> userId: GAMrfHxCQnHngnN7k, arguments: [{}]
```

**Comment by urosino** [comment]
Since nobody commented so far, does editing message work to everyone else, running 3.17.0?

**Comment by debdutdeb** [comment]
Hi Uros, thanks for reporting this.

> does editing message work for everyone else, running 3.17.0?  

Yes, it does in my test servers and also on open. 

![image](https://user-images.githubusercontent.com/76006232/128044691-8b11d219-30e7-4e72-8930-b6b4e10b3da0.png)

I suspect the issue is not exactly about editing a message. I'll have to dig deeper, there are couple other issues that are surfacing, are of the same type.  

For now, does it happen all the time? What happens if you restart the container?

**Comment by urosino** [comment]
@debdutdeb even after restarting the container, the issue persists.

```
server.js:204 Integrations ➔ Outgoing WebHook.debug Starting search for triggers for the room: EkJFRSn2natkMzooF
server.js:204 Integrations ➔ Outgoing WebHook.debug Found 0 to iterate over and see if the match the event.
server.js:204 Meteor ➔ method readMessages -> userId: 4xbX6cdmkMuqkCxF8, arguments: ["EkJFRSn2natkMzooF"]
server.js:204 Meteor ➔ method updateMessage -> userId: 4xbX6cdmkMuqkCxF8, arguments: [{"_id":"wserrDgRkyzD8BA8f","rid":"EkJFRSn2natkMzooF"}]
Exception while invoking method 'updateMessage' TypeError: Cannot read property 't' of undefined
    at isOmnichannelRoom (definition/IRoom.ts:96:90)
    at ee/app/canned-responses/server/hooks/onMessageSentParsePlaceholder.ts:32:7
    at callbacks.runItem (app/callbacks/lib/callbacks.js:111:70)
    at Object.callbacks.runItem (app/metrics/server/callbacksMetrics.js:24:20)
    at app/callbacks/lib/callbacks.js:39:35
    at app/callbacks/lib/callbacks.js:45:45
    at callbacks.run (app/callbacks/lib/callbacks.js:127:9)
    at Object.callbacks.run (app/metrics/server/callbacksMetrics.js:14:17)
    at updateMessage (app/lib/server/functions/updateMessage.js:48:22)
    at MethodInvocation.updateMessage (app/lib/server/methods/updateMessage.js:71:10)
    at MethodInvocation.methodsMap.<computed> (app/lib/server/lib/debug.js:76:34)
    at maybeAuditArgumentChecks (packages/ddp-server/livedata_server.js:1771:12)
    at packages/ddp-server/livedata_server.js:719:19
    at Meteor.EnvironmentVariable.EVp.withValue (packages/meteor.js:1234:12)
    at packages/ddp-server/livedata_server.js:717:46
    at Meteor.EnvironmentVariable.EVp.withValue (packages/meteor.js:1234:12)
    at packages/ddp-server/livedata_server.js:715:46
    at new Promise (<anonymous>)
    at Session.method (packages/ddp-server/livedata_server.js:689:23)
    at packages/ddp-server/livedata_server.js:559:43
server.js:204 Meteor ➔ method UserPresence:away -> userId: hvMn7DEsBdHthBNvJ, arguments: [{}]
server.js:204 Meteor ➔ method UserPresence:away -> userId: zoKTzFPXqsWQLLwHd, arguments: [{}]
server.js:204 Meteor ➔ method UserPresence:away -> userId: igskPjDtTsyWgyHRq, arguments: [{}]
server.js:204 Meteor ➔ method UserPresence:online -> userId: e5iZbSuXtb3jEDe3G, arguments: [{}]
```

**Comment by debdutdeb** [comment]
> at isOmnichannelRoom (definition/IRoom.ts:96:90)

Where are you trying to edit the message? I.e. in a private channel, dm, public channel or omnichannel?

**Comment by urosino** [comment]
Any room, it's not related to any specific type of channel.

**Comment by urosino** [comment]
@debdutdeb do you maybe have any tips how to debug and solve that problem?

As it looks, it's isolated to our setup.

**Comment by debdutdeb** [comment]
> it's isolated to our setup.

Not anymore. Someone else opened a similar issue yesterday. Sorry for the late response, I'll update both of you asap.

**Comment by Montel** [comment]
I confirm this bug.
We update 3.17.0 and now we can't edit message.
It's a critical bug.

**Comment by Montel** [comment]
I implemented a RC client and I use a rest api.
it returns:
"Cannot read property 't' of undefined" when I try to edit message

**Comment by Montel** [comment]
Do you need specific rest api call ?

**Comment by steffen-kdab** [comment]
We have a bunch of different RC server instances running, and the only one that has this problem is the only one that has OmniChannel enabled.

`| Exception while invoking method 'updateMessage' TypeError: Cannot read property 't' of undefined
|     at isOmnichannelRoom (definition/IRoom.ts:96:90)
|     at ee/app/canned-responses/server/hooks/onMessageSentParsePlaceholder.ts:32:7
|     at callbacks.runItem (app/callbacks/lib/callbacks.js:111:70)
|     at Object.callbacks.runItem (app/metrics/server/callbacksMetrics.js:24:20)
|     at app/callbacks/lib/callbacks.js:39:35
...
`

**Comment by urosino** [comment]
I also need to confirm we have the Enterprise license if this may be related.

**Comment by debdutdeb** [comment]
Thanks Montel & Steffen for your comments 🙂 

I have notified our product manager, and he'll be looking at it VERY soon. 

> I also need to confirm we have the Enterprise license if this may be related.

Thanks Uros. I'll pass this along as well.

Unfortunately I haven't got the time to look and investigate this issue personally. The product team will delegate this to the appropriate people asap.

**Comment by steffen-kdab** [comment]
Our server instance that has the problem indeed has the enterprise license. The ones that work fine are community licensed.

**Comment by debdutdeb** [comment]
Ahh! Stefan thanks!! That really helps!

**Comment by urosino** [comment]
@debdutdeb have you found what is triggering this bug? I really hope this can be resolved soon.

**Comment by debdutdeb** [comment]
I personally haven't gotten the time urosino, really sorry. But I already have notified our product manager.

You will get some concrete update on this tomorrow, 100%. Because I have a meeting with the product team tomorrow **AND** you have an enterprise license. Just a bit more patience - I know not being able to edit messages can be frustrating, especially when it used to work just a couple of days ago.

**Comment by debdutdeb** [comment]
I personally haven't gotten the time urosino, really sorry. But I already have notified our product manager.

I have a meeting with the product team tomorrow, you will get a concrete update on this then, 100%. Being on enterprise edition will help as well. Just a bit more patience - I know not being able to edit messages can be frustrating, especially when it used to work just a couple of days ago.

**Comment by yashovardhan** [comment]
Hey there,

We had a conversation with the product team about this. This issue is already highlighted there and they're working on it. Most probably, it will be fixed in the next release.

**Comment by Montel** [comment]
Hi,
Ok great. But I hope that you will release soon a next release as it's a very critical bug.
Thanks

**Comment by yashovardhan** [comment]
Hey Montel,

We have monthly releases. So you can expect this fixed by the September/ October release.

**Comment by steffen-kdab** [comment]
Great work! Message editing works again in 3.17.1 / Enterprise.

**Comment by debdutdeb** [comment]
Fixed in 3.17.1

Closing :)
